src/pages/LoginPage/LoginPage.tsx

```diff
@@ -21,14 +29,56 @@ const LoginPage: React.FC = () => {
   const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const { name, value } = e.target;
     setAuthenticationRequest({ ...authenticationRequest, [name]: value });
+    setValidationErrors({ ...validationErrors, [name]: false });
+  };
+
+  const validateInput = () => {
+    let isValid = true;
+
+    Object.keys(authenticationRequest).forEach((key) => {
+      const fieldName = key as keyof AuthenticationRequest;
+      const isFieldValid = authenticationRequest[fieldName].trim() !== '';
+
+      setValidationErrors((prevErrors) => ({
+        ...prevErrors,
+        [fieldName]: !isFieldValid
+      }));
+
+      if (!isFieldValid) {
+        isValid = false;
+      }
+    });
+
+    if (!isValid) {
+      setErrorMessage('Please fill in all fields.');
+    }
+
+    return isValid;
   };
 
   const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
 
-    authenticate(authenticationRequest);
-
-    setAuthenticationRequest({ username: '', password: '' });
+    if (!validateInput()) {
+      return;
+    } else {
+      authenticate(authenticationRequest)
+        .then((response) => {
+          if (response.status == 200) {
+            localStorage.setItem('token', JSON.stringify(response.data.token));
+            navigate('/home');
+          }
+        })
+        .catch((error) => {
+          console.log(error);
+          if (error.response.status == 403) {
+            setErrorMessage('Wrong credentials!');
+          }
+          if (error.response.status == 404) {
+            setErrorMessage(error.response.data.message);
+          }
+        });
+    }
   };
```

Comment:
Why having double condition? You can omit having if else and can just write part of code in else if input is valid